weblate/vcs/git.py

```diff
@@ -759,54 +792,68 @@ class GitLabRepository(GitMergeRequestBase):
 
     _version = None
 
-    # docs: https://zaquestion.github.io/lab/
-    _cmd = "lab"
-
-    @classmethod
-    def _get_version(cls):
-        """Return VCS program version."""
-        try:
-            return cls._popen(["--version"], merge_err=False).split()[-1]
-        except RepositoryException as error:
-            # It asks for configuration even with --version, see
-            # https://github.com/zaquestion/lab/issues/374
-            if error.retcode == 1 and "EOF" in error.get_message():
-                return "0.16"
-            raise
-
     @staticmethod
     def get_username():
         return settings.GITLAB_USERNAME
 
-    def create_pull_request(self, origin_branch, fork_remote, fork_branch):
-        """Create merge (a.k.a pull) request.
+    def api_url(self):
+        repo_name_index = self.component.repo.rfind(
+            "/", 0, self.component.repo.rfind("/")
+        )
+        api_url = "{}/api/v4/projects/{}".format(
+            self.component.repo[:repo_name_index],
+            urllib.parse.quote(
+                self.component.repo[(repo_name_index + 1) :].replace(".git", ""),
+                safe="",
+            ),
+        )
+        return api_url
+
+    def create_fork(self):
+        get_fork_url = "{}/forks?owned=True".format(self.api_url())
+        fork_url = "{}/fork".format(self.api_url())
+        ssh_url_to_forked_repo = ""
+
+        # Check if Fork already exists owned by current user. If the
+        # fork already exists, set that fork as remote.
+        # Else, create a new fork
+        r = requests.get(
+            get_fork_url,
+            headers={"Authorization": "Bearer {}".format(settings.GITLAB_TOKEN)},
+        )
+        for fork in r.json():
+            if "owner" in fork and fork["owner"]["username"] == self.get_username():
```

Comment:
Shouldn't the owned=True condition return only the owned ones making this condition not needed? 

In case it doesn't filter that, we have scalability issue as projects can easily have thousands of forks...

Reply:
It just returns the forks owned by users and any forks in a group owned by the user. So the responses will be restricted to only user's profile and all the groups, but we need that condition to identify the fork in the user's profile and not in the groups. I will add comments explaining the same.